pkg/operator/ceph/csi/spec.go

```diff
@@ -470,18 +481,75 @@ func startDrivers(namespace string, clientset kubernetes.Interface, ver *version
 	}
 
 	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor >= provDeploymentSuppVersion) {
-		err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+		if EnableRBD {
+			err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+			}
 		}
-		err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+		if EnableCephFS {
+			err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+			}
+		}
+	}
+	return nil
+}
+
+func stopDrivers(namespace string, clientset kubernetes.Interface) error {
+	if !EnableRBD {
+		logger.Info("CSI Ceph RBD driver disabled ")
+		succeeded := deleteCSIDriverResources(namespace, clientset, RBDPlugin, RBDProvisioner, "csi-rbdplugin-metrics", RBDDriverName)
+		if succeeded {
+			logger.Info("successfully removed CSI Ceph RBD driver")
+		} else {
+			logger.Info("failed to remove CSI Ceph RBD driver")
```

Comment:
Why not returning an error? Then the caller can decide what to do with it

Reply:
> Why not returning an error? Then the caller can decide what to do with it

If we'll return it will stop deleting other resources. We have discussed in [this](https://github.com/rook/rook/pull/5605#discussion_r446282231) and [this](https://github.com/rook/rook/pull/5605#discussion_r446282692)

Reply:
ok so at least make it Error instead of Info